websites/G/Genshin Impact Map/presence.ts

```diff
@@ -0,0 +1,147 @@
+const presence = new Presence({
+		clientId: "973729201104511007",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+interface Maps {
+	city: boolean;
+	id: number;
+	map: string;
+	largeImageKey: string;
+	pvlargeImageKey: string | null;
+	smallImageKey: string;
+}
+
+interface City {
+	position?: number;
+	map: string;
+	largeImageKey: string;
+	smallImageKey: string;
+}
+
+const map: Maps[] = [
+		{
+			city: true,
+			id: 2,
+			map: "Teyvat",
+			largeImageKey: "teyvat_map",
+			pvlargeImageKey: "teyvat_map",
+			smallImageKey: "emblem_unknown",
+		},
+		{
+			city: false,
+			id: 7,
+			map: "Enkanomiya",
+			largeImageKey: "enkanomiya_map",
+			pvlargeImageKey: "preview_enkanomiya",
+			smallImageKey: "emblem_enkanomiya",
+		},
+		{
+			city: false,
+			id: 9,
+			map: "The Chasm: Underground Mines",
+			largeImageKey: "the_chasm_underground_mines_map",
+			pvlargeImageKey: "preview_the_chasm_underground_mines",
+			smallImageKey: "emblem_thechasm",
+		},
+		{
+			city: false,
+			id: 0,
+			map: "Unknown",
+			largeImageKey: "unknown_map",
+			pvlargeImageKey: null,
+			smallImageKey: "emblem_unknown",
+		},
+	],
+	city: City[] = [
+		{
+			position: 1200,
+			map: "Mondstadt",
+			largeImageKey: "preview_mondstadt",
+			smallImageKey: "emblem_mondstadt",
+		},
+		{
+			position: 4000,
+			map: "Liyue",
+			largeImageKey: "preview_liyue",
+			smallImageKey: "emblem_liyue",
+		},
+		{
+			position: 9000,
+			map: "Inazuma",
+			largeImageKey: "preview_tenshukaku",
+			smallImageKey: "emblem_inazuma",
+		},
+	];
+
+let getpos: number, current: Maps, currentCity: City;
+
+presence.on("UpdateData", async () => {
+	const [showPreview, timestamps] = await Promise.all([
+			presence.getSetting<boolean>("showPreview"),
+			presence.getSetting<boolean>("timestamps"),
+		]),
+		presenceData: PresenceData = {
+			details: "Genshin Impact Map",
+			largeImageKey: "main",
+			smallImageKey: "search",
+			startTimestamp: browsingTimestamp,
+		},
+		{ host, hostname, href } = document.location;
```

Comment:
Seems like you didn’t get the hint so I’ll be more specific. There’s no reason for you to use the href here. You should be using search and pathname